main.go

```diff
@@ -132,77 +136,76 @@ func getProjectID(baseURL string, token string, projectname string, namespace st
 }
 
 // Get and return currently active milestones
-func getMilestones(baseURL string, token string, projectID string) (map[string]string, error) {
-	milestones := []milestoneAPI{}
-
-	strURL := []string{baseURL, projectID, "/milestones?state=active&per_page=100"}
-	URL := strings.Join(strURL, "")
-
-	m, linkHeader, err := callServer(URL, token, milestones)
-	if err != nil {
-		return m, err
-	}
+func getActiveMilestones(baseURL string, token string, projectID string) ([]milestoneAPI, error) {
+	state := "active"
+	return getMilestones(baseURL, token, projectID, state)
+}
 
-	for _, milestone := range milestones {
-		m[milestone.Title] = milestone.DueDate
-	}
+// Get and return inactive milestones
+func getInactiveMilestones(baseURL string, token string, projectID string) ([]milestoneAPI, error) {
+	state := "closed"
+	return getMilestones(baseURL, token, projectID, state)
+}
 
-	parsedHeader := link.Parse(linkHeader)
-	for _, elem := range parsedHeader {
-		if elem.Rel == "next" {
-			m, _, err := callServer(URL, token, milestones)
-			if err != nil {
-				return m, err
+func getMilestones(baseURL string, token string, projectID string, state string) ([]milestoneAPI, error) {
+	m := []milestoneAPI{}
+	client := &http.Client{}
+	strURL := []string{baseURL, "/projects/", projectID, "/milestones"}
+	URL := strings.Join(strURL, "")
+	params := url.Values{}
+	params.Add("state", state)
+	paginate := true
+	for paginate == true {
+		paginate = false
+		req, err := http.NewRequest("GET", URL, strings.NewReader(params.Encode()))
+		if err != nil {
+			return m, err
+		}
+		req.Header.Add("PRIVATE-TOKEN", token)
+		resp, err := client.Do(req)
+		if err != nil {
+			return m, err
+		}
+		respByte, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return m, err
+		}
+		json.Unmarshal(respByte, &m)
+		defer resp.Body.Close()
+		linkHeader := resp.Header.Get("Link")
+		parsedHeader := link.Parse(linkHeader)
+		var pages int
+		links := map[int]string{}
+		for _, elem := range parsedHeader {
+			if elem.Rel != "next" {
+				continue
 			}
-			for _, milestone := range milestones {
-				m[milestone.Title] = milestone.DueDate
+			if elem.Rel == "next" {
+				pages++
+				link := elem.URI
+				links[pages] = link
 			}
 		}
+		for k := range links {
+			URL = links[k]
+			paginate = true
+			break
+		}
```

Comment:
What was your thought behind this code? Within the outer loop, which basically requests URL in every run, you have this inner loop, which ranges over the number of links and prints the first one and then breaks.